parser_library/src/context/instruction.h

```diff
@@ -334,6 +335,73 @@ class inline_string
     }
 };
 
+enum class condition_code
+{
+    _0,
+    _1,
+    _2,
+    _3,
+};
+
+class condition_code_explanation
+{
+    std::array<const char*, 5> text;
+    std::array<unsigned char, 5> lengths;
+
+public:
+    template<size_t l0>
```

Comment:
Consider renaming the template parameter to something else. I've got confused for a bit as it looks really similar to a number (`l0` ~ `10`). I suggest the following:
```suggestion
    template<size_t L0>
```

The same comment goes for other two template definitions below.